tree/20_cfe_basics/services.cf

```diff
@@ -257,33 +257,33 @@ bundle agent ncf_services(service, action)
 
   methods:
     pass1.!pass2::
-        "${report_data.method_id}_${service}" usebundle => _classes_cancel("${report_data.method_id}");
+        "${report_data.method_id}_${service}" usebundle => _classes_cancel("${report_data.report_id}");
     #####
     # Actual command - for actions, non using systemd
     #####
     pass2.is_check_action.!is_process_action.method_found.action_ok::
       "${report_data.method_id}" usebundle => _classes_success("${class_prefix}");
-      "${report_data.method_id}" usebundle => _classes_success("${report_data.method_id}");
+      "${report_data.method_id}" usebundle => _classes_success("${report_data.report_id}");
```

Comment:
this won't allow making a difference in methods calling several other methods.

Reply:
all will use the same prefix, method_id is there for a reason

Reply:
method_id and report_id were using the same value

Reply:
"report_data.report_id"      string => canonify("${report_id}_${report_data.directive_id}");
    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");

Reply:
So if you have the problem with report_id you had the problem with method_id

Hence this pr makes method id even more specific (and i keep use it as promiser because of this) as it allow reexecution of things that were not reexecuted

Reply:
Anyway i made this changed in 7.2 but i think it should be on master

Reply:
method_id is modified each time we enter a sub-method (by the push/pop mechanism), report_id stays the same. What is in _method_reporting_context_v4 is only the initialization.